uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -52,10 +52,14 @@ define([
           public_holiday: isPublicHoliday || false
         };
 
-        deferred.resolve(this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params, false)
+        this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params)
           .then(function (data) {
-            return data.values;
-          }));
+            if(data && data.is_error == 1){
```

Comment:
1. `if(data && !!data.is_error) {`
2. Leave a space before "{"
3. We should not send the error object via `reject`, instead we can just send  
`deferred.reject(data.error_message)`;